### ISSUE TITLE
Issue decompiling from PowerPC ASM generated by IDA Pro

### ISSUE THREAD

**Issue by kernaltrap8** [matt-kempster/m2c#264]
Hello,
I disassembled a ELF from a Playstation 3 and exported it as ASM, however when running it through m2c it gives me this error:
```
PS C:\cygwin\home\693982\m2c> python3 m2c.py ..\WoRmod\binary\src\EBOOT.asm --target ppc-mwcc-c
/*
Decompilation failure:

unsupported non-nop instruction outside of function (OPDEntry struc)
*/
```

I don't know why this isn't working. Does m2c not support AltiVec PowerPC?

**Comment by JamesIsWack** [comment]
ah i see. thank you for replying. let me know when AltiVec support is added.

**Comment by simonlindholm** [comment]
Can you post the asm you're trying? The issue you're hitting isn't about altivec, it looks like a formatting thing. I imagine altivec instructions won't occur very commonly so m2c might work alright most of the time (although probably not as well as ghidra; we haven't put too much effort into ppc support).

**Comment by JamesIsWack** [comment]
> Can you post the asm you're trying? The issue you're hitting isn't about altivec, it looks like a formatting thing. I imagine altivec instructions won't occur very commonly so m2c might work alright most of the time (although probably not as well as ghidra; we haven't put too much effort into ppc support).
its a whole program, about 250MB of assembly. i can upload it and send it here, if you like. its just the direct dump from the ELF in IDA Pro.

**Comment by simonlindholm** [comment]
can you try to run m2c on just the first function of it, and post that here if it doesn't work?

**Comment by simonlindholm** [comment]
Yes, that works. But I was thinking you could reduce the .asm file to include just a single function, to have something to post here that's easier to work with and less infringing on copyright.
